### ISSUE TITLE
Subqueries using LET not working as expected when LUCENE index involved

### ISSUE THREAD

**Issue by misu200** [orientechnologies/orientdb#8521]
### OS: Ubuntu 16.04

When the WHERE part of a SELECT is using a LUCENE index the context variables from LET not working as expected. 

I ve created the following simple database test:

```
create class A extends V;
create property A.name  string;
create index A.nameLucene on A(name) fulltext engine LUCENE;
create index A.nameHash ON A (name) NOTUNIQUE_HASH_INDEX;
		   
create class B extends V;
create class X extends E;								  
let $a = create vertex A set name="Mike";
let $b = create vertex B;
create edge X from $a to $b;
```

When using a LUCENE index the $t is always empty ( [] ) :
`select $t,* from A let $t=(select * from (select expand($parent.$parent.$current.out()))) where SEARCH_CLASS("Mike") = true;`

Similar query but replacing LUCENE with a  different index type returns the expected non empty value for $t :
`select $t,* from A let $t=(select * from (select expand($parent.$parent.$current.out()))) where name="Mike";`

Thanks,
Mihai

**Comment by luigidellaquila** [comment]
Hi @misu200 

Thank you very much for reporting, I managed to reproduce the problem. 
I'm testing a fix now, I'll let you know ASAP

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @misu200 

I just pushed a fix on 3.0.x branch, it will be released with V 3.0.8

Thanks

Luigi
